src/hotspot/cpu/x86/x86.ad

```diff
@@ -2048,9 +2099,20 @@ const bool Matcher::match_rule_supported_vector_masked(int opcode, int vlen, Bas
       }
       return true;
 
+    case Op_PopCountVI:
+    case Op_PopCountVL:
+      if (!is_pop_count_instr_target(bt)) {
+        return false;
+      }
+      return true;
+
     case Op_MaskAll:
       return true;
 
+    case Op_CountLeadingZerosV:
+      if ((bt == T_INT || bt == T_LONG) && VM_Version::supports_avx512cd()) {
```

Comment:
Newly introduced `is_non_subword_integral_type(bt)` can be used here instead of `bt == T_INT || bt == T_LONG`.